qutebrowser/javascript/webengine_caret.js

```diff
@@ -0,0 +1,1104 @@
+                if (cursor.node.constructor !== Text) {
+                    nodesCrossed.push(cursor.node);
+                }
+            } else {
+                if (cursor.index < cursor.text.length) {
+                    return cursor.text[cursor.index++];
+                }
+
+                while (cursor.node !== null) {
+                    let siblingNode = null;
+                    for (let node = cursor.node.nextSibling;
+                        node !== null;
+                        node = node.nextSibling) {
+                        if (TraverseUtil.isSkipped(node)) {
+                            nodesCrossed.push(node);
+                        } else if (TraverseUtil.isVisible(node)) {
+                            siblingNode = node;
+                            break;
+                        }
+                    }
+                    if (siblingNode) {
+                        cursor.node = siblingNode;
+                        cursor.text = TraverseUtil.getNodeText(siblingNode);
+                        cursor.index = 0;
+
+                        if (cursor.node.constructor !== Text) {
+                            nodesCrossed.push(cursor.node);
+                        }
+
+                        break;
+                    }
+
+                    const parentNode = cursor.node.parentNode;
+                    if (parentNode &&
+                        parentNode.constructor !== HTMLBodyElement) {
+                        cursor.node = cursor.node.parentNode;
+                        cursor.text = null;
+                        cursor.index = 0;
+                    } else {
+                        return null;
+                    }
+                }
+            }
+        }
+    };
+
+    TraverseUtil.getNextChar = function( // eslint-disable-line max-params
+        startCursor, endCursor, nodesCrossed, skipWhitespace) {
+        startCursor.copyFrom(endCursor);
+        let fChar = TraverseUtil.forwardsChar(endCursor, nodesCrossed);
+        if (fChar === null) {
+            return null;
+        }
+
+        const initialWhitespace = TraverseUtil.isWhitespace(fChar);
+
+        while ((TraverseUtil.isWhitespace(fChar)) ||
+            (TraverseUtil.isSkipped(endCursor.node))) {
+            fChar = TraverseUtil.forwardsChar(endCursor, nodesCrossed);
+            if (fChar === null) {
+                return null;
+            }
+        }
+        if (skipWhitespace || !initialWhitespace) {
+            startCursor.copyFrom(endCursor);
+            startCursor.index--;
+            return fChar;
+        }
+
+        for (let i = 0; i < nodesCrossed.length; i++) {
+            if (TraverseUtil.isSkipped(nodesCrossed[i])) {
+                endCursor.index--;
+                startCursor.copyFrom(endCursor);
+                startCursor.index--;
+                return " ";
+            }
+        }
+        endCursor.index--;
+        return " ";
+    };
+
+    const CaretBrowsing = {};
```

Comment:
Why did you remove the comments (here and elsewhere) from the upstream file? Those seemed quite useful at understanding what's going on...

Reply:
The code looks self explanatory enough for me. I thought those comments too redundant. And with given the fact that js file became too big I decided to clean this comments. But of cause, if you think that this comments will help someone I can easily return it.